pkg/encryption/path.go

```diff
@@ -159,3 +164,59 @@ func decryptPathComponent(comp string, cipher storj.CipherSuite, key *storj.Key)
 
 	return string(decrypted), nil
 }
+
+func encodeSegment(segment []byte) []byte {
+	if len(segment) == 0 {
+		return emptyComponent
+	}
+
+	result := make([]byte, 0, len(segment)*2+1)
+	result = append(result, notEmptyComponentPrefix)
+	for i := 0; i < len(segment); i++ {
+		switch {
```

Comment:
You can change this to

```go
switch segment[i] {
case escape1:
case escape1+1:
case escape2:
...
```

which seems cleaner to me.

Reply:
good idea